sql-plugin/src/main/311+-all/scala/com/nvidia/spark/rapids/shims/v2/GpuRowBasedScalaUDF.scala

```diff
@@ -74,6 +74,17 @@ case class GpuRowBasedScalaUDF(
       (CatalystTypeConverters.createToScalaConverter(dataType), false)
     }
   }
+
+  /**
+   *  Need nulls check when there are array types with nulls in the input.
+   *  This is for `https://github.com/NVIDIA/spark-rapids/issues/3942`.
+   */
+  override val checkNull: Boolean = children.exists { child =>
+    child.dataType match {
+      case ArrayType(_, containsNull) => containsNull
```

Comment:
This looks like it is only checking for a top-level array? Do we need to recurse into other types to check for nested arrays?

Reply:
Nice catch. will update it.

Reply:
Updated